modules/experimental-layers/src/addons/json-layer/get.js

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2015 - 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+/**
+ * Access properties of nested containers using dot-path notation
+ * - Supports plain objects and arrays, as well as classes with `get` methods
+ *   such as ES6 Maps, Immutable.js objects etc
+ * - Returns undefined if any container is not valid, instead of throwing
+ *
+ * @param {Object} container - container that supports get
+ * @param {String|*} compositeKey - key to access, can be '.'-separated string
+ * @return {*} - value in the final key of the nested container
+ */
+export function get(container, compositeKey) {
+  // Split the key into subkeys
+  const keyList = getKeys(compositeKey);
```

Comment:
> The keyMap solution breaks the closure and creates yet another global store. IMO this is the place where "get getter" functions make sense:

I feel that I might not have fully understood your comment, but I feel the global `keyMap` is there to mitigate the concern of having to `split('.')` the accessor strings on every render (since we have to mint new getters every frame). 

If we move the support for string accessors upstream into core, then we could defer the splitting until we actually need these accessors in the AttributeManager, and then we could easily take the splitting cost as it would be minimal compared to attribute update. 

If not, maybe I should add some benchmarks to justify/refute the value of the added complexity here?

Reply:
In my mind it's more important to optimize this for attribute generation - when `getKeys` is called for every data object. Before we removed this in v5, it created hundreds of thousands of new key arrays  whenever attributes are recalculated.